File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.swift
import Foundation

func solution(_ sequence:[Int], _ k:Int) -> [Int] {
    var st = 0
    var en = 0
    var ans = [0, sequence.count - 1]
    var sum = sequence[st]
    while en < sequence.count {
        if sum == k {
            if en-st < ans[1]-ans[0] {
                ans = [st, en]
            }
            sum -= sequence[st]
            st += 1
        } else if sum < k {
            en += 1
            if en == sequence.count { break }
            sum += sequence[en]
        } else {
            sum -= sequence[st]
            st += 1
        }
    }
    return ans
}

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.java
class Solution {
    public int[] solution(int[] sequence, int k) {
        int[] answer = new int[2];
        int minlen = Integer.MAX_VALUE;
        
        int left = 0, right = 0, sum = 0;
        while(right < sequence.length){
            sum += sequence[right];
            
            while (sum > k) {
                sum -= sequence[left++];
            }
            if(sum == k) {
                int len = right - left;
                if (minlen > len) {
                    minlen = len;
                    answer[0] = left;
                    answer[1] = right;
                }
            }
            right++;
        }
        return answer;
    }
}